shopfloor_base_multicompany/services/service.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 Camptocamp SA (http://www.camptocamp.com)
+# @author Simone Orsi <simahawk@gmail.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+from odoo import _, exceptions
+
+from odoo.addons.component.core import AbstractComponent
+
+
+class BaseShopfloorService(AbstractComponent):
+
+    _inherit = "base.shopfloor.service"
+
+    def dispatch(self, method_name, *args, params=None):
+        self._validate_company()
+        return super().dispatch(method_name, *args, params=params)
+
+    def _validate_company(self):
+        """
+        Validate current user's companies match one of the companies allowed for the app.
+
+        Raises:
+        exceptions.UserError: If the user's company is not allowed.
+        """
+        if self.collection.must_validate_company:
+            if not self.env.user.company_ids & self.collection.company_ids:
```

Comment:
All new records created from the users in the app will likely belong to the company of the user. So we need to ensure a user active in multiple companies is logged in the right company. 
```suggestion
            if not self.env.user.company_id in self.collection.company_ids:
```

Reply:
fixed